src/apps/Hosting/HostingDialog.js

```diff
@@ -19,23 +19,39 @@ const CreateHostingDialog = React.createClass({
     FormMixin
   ],
 
-  validatorConstraints: {
-    name: {
-      presence: true,
-      exclusion: {
-        within: ['default', 'Default'],
-        message: "can't be a 'default'"
+  validatorConstraints() {
+    const { cname, name } = this.state;
+    const isCnameValid = /\.[a-zA-Z]+$/.test(cname);
+    const validateObj = {
+      name: {
+        presence: true,
+        format: {
+          pattern: '[a-zA-Z0-9-_]+$',
+          message: 'can contain only a-z, 0-9, no spaces '
+        }
       },
-      format: {
-        pattern: '[a-zA-Z0-9-_]+$',
-        message: 'can contain only a-z, 0-9, no spaces '
-      }
-    },
-    description: {
-      length: {
-        maximum: 256
+      description: {
+        length: {
+          maximum: 256
+        }
       }
+    };
+
+    if (!isCnameValid) {
+      validateObj.cname = {
+        inclusion: {
+          message: '^CNAME is not a valid url '
```

Comment:
I think that CNAME has kind of the same meaning as URL / ALIAS.  So probably domain should go here, but I am not an expert (yet ;p)